### ISSUE TITLE
[Emulator] Setting phone number through Update User does not work with Firebase Auth Emulator

### ISSUE THREAD

**Issue by stevenolay** [firebase/firebase-tools#3180]
### [READ] Step 1: Are you in the right place?

I believe I am, tried my best, apologies in advanced. I searched for other emulator related issues in the repository first.

### [REQUIRED] Step 2: Describe your environment

  * Operating System version: MacOS
  * Firebase SDK version: 9.5.0
  * Firebase Product: Firestore
  * Node.js version: 12.18.3
  * NPM version: 6.14.6

### [REQUIRED] Step 3: Describe the problem
If my Firebase admin SDK is pointed to the Firestore auth emulator, when attempting to set the phone number on a UserRecord using the updateUser method the user is not updated with the phone number. If I point the Firebase admin SDK upstream to one of my Firebase environments and run the same code snippet, updateUser correctly sets the phone number onto the UserRecord.  
#### Steps to reproduce:

What happened? How can we make the problem occur?
This could be a description, log/console output, etc.

Code snippet: 
![image](https://user-images.githubusercontent.com/9903968/109571425-b0e0e800-7ab9-11eb-8481-af42e9a6bd84.png)

Result when pointed to auth emulator:
```
UserRecord {
  uid: 'ooEjXg58tdLgsZUvHwEb2WEVkqSw',
  email: undefined,
  emailVerified: false,
  displayName: undefined,
  photoURL: undefined,
  phoneNumber: undefined,
  disabled: false,
  metadata: UserMetadata {
    creationTime: 'Mon, 01 Mar 2021 22:53:27 GMT',
    lastSignInTime: 'Mon, 01 Mar 2021 22:53:27 GMT',
    lastRefreshTime: null
  },
  providerData: [],
  passwordHash: undefined,
  passwordSalt: undefined,
  tokensValidAfterTime: undefined,
  tenantId: undefined
}
UserRecord {
  uid: 'ooEjXg58tdLgsZUvHwEb2WEVkqSw',
  email: undefined,
  emailVerified: false,
  displayName: undefined,
  photoURL: undefined,
  phoneNumber: undefined,
  disabled: false,
  metadata: UserMetadata {
    creationTime: 'Mon, 01 Mar 2021 22:53:27 GMT',
    lastSignInTime: 'Mon, 01 Mar 2021 22:53:27 GMT',
    lastRefreshTime: null
  },
  providerData: [],
  passwordHash: undefined,
  passwordSalt: undefined,
  tokensValidAfterTime: undefined,
  tenantId: undefined
}
```

Result when pointed upstream to a true Firestore environment:
```
UserRecord {
  uid: '6MmhqyOqLvNXlPUSCE68S59OkHz2',
  email: undefined,
  emailVerified: false,
  displayName: undefined,
  photoURL: undefined,
  phoneNumber: undefined,
  disabled: false,
  metadata: UserMetadata {
    creationTime: 'Mon, 01 Mar 2021 22:56:40 GMT',
    lastSignInTime: null,
    lastRefreshTime: null
  },
  providerData: [],
  passwordHash: undefined,
  passwordSalt: undefined,
  tokensValidAfterTime: 'Mon, 01 Mar 2021 22:56:40 GMT',
  tenantId: undefined
}
UserRecord {
  uid: '6MmhqyOqLvNXlPUSCE68S59OkHz2',
  email: undefined,
  emailVerified: false,
  displayName: undefined,
  photoURL: undefined,
  phoneNumber: '+16505551111',
  disabled: false,
  metadata: UserMetadata {
    creationTime: 'Mon, 01 Mar 2021 22:56:40 GMT',
    lastSignInTime: null,
    lastRefreshTime: null
  },
  providerData: [
    UserInfo {
      uid: '+16505551111',
      displayName: undefined,
      email: undefined,
      photoURL: undefined,
      providerId: 'phone',
      phoneNumber: '+16505551111'
    }
  ],
  passwordHash: undefined,
  passwordSalt: undefined,
  tokensValidAfterTime: 'Mon, 01 Mar 2021 22:56:40 GMT',
  tenantId: undefined
}
```
The provider data fails to update when pointed to the emulator, but works just fine when pointed upstream.

#### Relevant Code:

```
  const admin = require('firebase-admin');
  admin
    .auth()
    .createUser({})
    .then(async user => {
      console.log(user);
      const updated = await admin
        .auth()
        .updateUser(user.uid, { phoneNumber: '+16505551111' });
      console.log(updated);
    });
```

**Comment by samhorlbeck** [comment]
Hi @stevenolay, your code snippet seems to work for me no problem. Here's the output from your snippet:

```js
UserRecord {
  uid: 'YbezMGddbPmb2i6N0xtWgBP3kuf8',
  email: undefined,
  emailVerified: false,
  displayName: undefined,
  photoURL: undefined,
  phoneNumber: undefined,
  disabled: false,
  metadata: UserMetadata {
    creationTime: 'Tue, 23 Mar 2021 15:33:25 GMT',
    lastSignInTime: 'Tue, 23 Mar 2021 15:33:25 GMT',
    lastRefreshTime: null
  },
  providerData: [],
  passwordHash: undefined,
  passwordSalt: undefined,
  tokensValidAfterTime: undefined,
  tenantId: undefined
}
UserRecord {
  uid: 'YbezMGddbPmb2i6N0xtWgBP3kuf8',
  email: undefined,
  emailVerified: false,
  displayName: undefined,
  photoURL: undefined,
  phoneNumber: '+16505551111',
  disabled: false,
  metadata: UserMetadata {
    creationTime: 'Tue, 23 Mar 2021 15:33:25 GMT',
    lastSignInTime: 'Tue, 23 Mar 2021 15:33:25 GMT',
    lastRefreshTime: null
  },
  providerData: [
    UserInfo {
      uid: '+16505551111',
      displayName: undefined,
      email: undefined,
      photoURL: undefined,
      providerId: 'phone',
      phoneNumber: '+16505551111'
    }
  ],
  passwordHash: undefined,
  passwordSalt: undefined,
  tokensValidAfterTime: undefined,
  tenantId: undefined
}
```

I know there's been a delay in our response to this bug (sorry about that). Would you mind double checking whether or not you're still seeing this behavior?

**Comment by stevenolay** [comment]
Just got around to looking at this again. Ended up having to fly blind for a little while. 

Looks like whatever the issue was, it has now since been resolved. I will try and write the test cases I needed again. Thanks for the update!

**Comment by stevenolay** [comment]
Tapping back in for others who might hit this. The issue I was having was I was not using the version of firebase-tools I thought I was. 

When running `firebase -V` I was getting 9.1.1 which is quite old. 

I tried to update, but i was still on nodejs 10 and it wouldn't go passed 9.1.1.

I switched to nodejs 12 and was able to update to 9.10.0.

The issue was completely resolved by moving to nodejs 12 and updating to the latest version of firebase-tools
